azext_iot/tests/iothub/state/test_hub_state_int.py

```diff
@@ -0,0 +1,957 @@
+import random
+import pytest
+import json
+import os
+import time
+from pathlib import Path
+from azext_iot.common.shared import DeviceAuthApiType
+
+from azext_iot.common.embedded_cli import EmbeddedCLI
+from azext_iot.tests.iothub.state.conftest import generate_hub_id
+from azext_iot.tests.settings import DynamoSettings, ENV_SET_TEST_IOTHUB_REQUIRED, ENV_SET_TEST_IOTHUB_OPTIONAL
+from azext_iot.tests.generators import generate_generic_id
+from azext_iot.common.utility import generate_key, read_file_content
+from azext_iot.common.certops import create_self_signed_certificate
+from azext_iot.tests.iothub import (
+    PRIMARY_THUMBPRINT,
+    SECONDARY_THUMBPRINT,
+    DATAPLANE_AUTH_TYPES,
+    DEVICE_TYPES,
+)
+from azext_iot.common._azure import parse_iot_hub_message_endpoint_connection_string, parse_storage_container_connection_string
+from azure.cli.core.azclierror import ResourceNotFoundError, RequiredArgumentMissingError
+
+settings = DynamoSettings(req_env_set=ENV_SET_TEST_IOTHUB_REQUIRED, opt_env_set=ENV_SET_TEST_IOTHUB_OPTIONAL)
+CWD = os.path.dirname(os.path.abspath(__file__))
```

Comment:
we have a fixture somewhere for this, `set_cwd` I think, that should automatically do this (or something similar) on any unit test method you pass it to. Not sure if it's the exact same functionality (or if it won't work because of scope, etc) but might be able to reuse it.

Reply:
The same code is littered in other tests too. It seems to be almost the same functionality but I would have to think about how things work with scope to make this work (since I will be assigning the dataplane role before doing all the dataplane uploads, which requires CWD)